File: cifrado_cesar/cifrado_cesar.pl
# Definir una constante para los desplazamientos
use constant SHIFT => 3;
 
# Se debe ingresar el texto plano 
my $texto = "Hola, Mundo!";
 
# Convertir el texto a minúsculas
$texto = lc($texto);
 
# Se inicializa el resultado
my $salida = "";
 
# Iterar sobre el texto para convertir 
# cada carácter
foreach my $char (split //, $texto) {
    # Desplazar el caracter a la derecha
    my $desp = ord($char) + SHIFT;
   
    # Validar que el desplazamiento de este caracter exista
    if ($desp > ord("z")) {
        # Volver el desplazamiento al rango
        $desp -= 26;
    }
 
    # Agregar el nuevo caracter al resultado
    $salida .= chr($desp);
}
 
# Imprimir el TextoCifrado
print $salida;

File: cifrado_cesar/cifrado_cesar.py
#Cifrador César en Python
def cifrar(texto, rotacion):  
    result = "" 
  
    # Recorrer todos los caracteres 
    for i in range(len(texto)): 
        char = texto[i] 
  
        # Minúsculas
        if (char.isupper()): 
            result += chr((ord(char) + rotacion-65) % 26 + 65) 
  
        # Mayúsculas 
        else: 
            result += chr((ord(char) + rotacion - 97) % 26 + 97) 
  
    return result 
  
# Verificación del programa 
texto = "Hola Mundo"
rotacion = 5

print("Texto original : " + texto) 
print("Rotación        : " + str(rotacion)) 
print("Texto cifrado   : " + cifrar(texto, rotacion)) 
